pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package mysql
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	// Add as per the documentation - https://github.com/go-sql-driver/mysql
+	_ "github.com/go-sql-driver/mysql"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// Provider represents an MySQL DB implementation of the storage.Provider interface
+type Provider struct {
+	dbPath string
+	db     *sql.DB
+	dbs    map[string]*sqlDBStore
+	sync.RWMutex
+}
+
+type sqlDBStore struct {
+	db        *sql.DB
+	tableName string
+}
+
+type result struct {
+	key   string
+	value []byte
+}
+
+const (
+	blankDBPathErrMsg         = "DB URL for new mySQL DB provider can't be blank"
+	failToCloseProviderErrMsg = "failed to close provider"
+	tablePrefix               = "t_"
+	sqlDBNotFound             = "no rows"
+	createDBQuery             = "CREATE DATABASE IF NOT EXISTS "
+	useDBQuery                = "USE "
+)
+
+var errKeyRequired = errors.New("key is mandatory")
+
+// NewProvider instantiates Provider
+func NewProvider(dbPath string) (*Provider, error) {
```

Comment:
dbURL would be better term than dbPath. Path usually refers to directories/files (unique location in a file system).